File: python/caesar_cipher.py
def caesar_cipher(string, shift_amount):
    lowercase = "abcdefghijklmnopqrstuvwxyz"
    uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    new_string = ""
    for ltr in string:
        if ltr in lowercase:
            index = lowercase.index(ltr)
            new_index = (index + shift_amount) % 26
            new_string += lowercase[new_index]
        elif ltr in uppercase:
            index = uppercase.index(ltr)
            new_index = (index + shift_amount) % 26
            new_string += uppercase[new_index]
        else:
            new_string += ltr

    return new_string

File: js/caesarCipher.js
const caesarCipher = function (string, shift_factor) {
  const uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
  const lowercase = "abcdefghijklmnopqrstuvwxyz";
  let newString = "";
  let newIndex;
  for (let ltr of string) {
    let lowercaseIndex = lowercase.indexOf(ltr);
    let uppercaseIndex = uppercase.indexOf(ltr);
    if (lowercaseIndex == -1 && uppercaseIndex == -1) {
      newString += ltr;
      continue;
    }
    if (lowercaseIndex != -1) {
      newIndex = lowercaseIndex + shift_factor;
      if (newIndex < 0) {
        newIndex = 26 + newIndex;
      }
      if (newIndex > 25) {
        newIndex %= 26;
      }
      newString += lowercase[newIndex];
      continue;
    }
    if (uppercaseIndex != -1) {
      newIndex = uppercaseIndex + shift_factor;
      if (newIndex < 0) {
        newIndex = 26 + newIndex;
      }
      if (newIndex > 25) {
        newIndex %= 26;
      }
      newString += uppercase[newIndex];
      continue;
    }
  }
  return newString;
};

module.exports = { caesarCipher };
